test/signer.ts

```diff
@@ -168,6 +168,108 @@ describe('signer', () => {
 
+      describe('usableFrom', () => {
+        it('should set correct settings if usableFrom provided', async () => {
+          const authClientSign: sinon.SinonStub<
+            [string],
+            Promise<string>
+          > = sandbox.stub(authClient, 'sign').resolves('signature');
+          const usableFrom = new Date(1581984000000);
+          await signer.getSignedUrl({
+            version: 'v4',
+            method: 'GET',
+            usableFrom,
+            expires: usableFrom.valueOf() + 5000,
+          });
+          const blobToSign = authClientSign.getCall(0).args[0];
+          assert(
+            blobToSign.includes(
+              dateFormat.format(usableFrom, 'YYYYMMDD[T]HHmmss[Z]', true)
+            )
+          );
+        });
+
+        it('should accept Date objects', async () => {
+          const usableFrom = new Date(1581984000678);
+          const signedUrl = await signer.getSignedUrl({
+            version: 'v4',
+            method: 'GET',
+            usableFrom,
+            expires: usableFrom.valueOf() + 5000,
+          });
+          const query = {
+            'X-Goog-Date': dateFormat.format(
+              usableFrom,
+              'YYYYMMDD[T]HHmmss[Z]',
+              true
+            ),
+          };
+          assert(signedUrl.includes(qsStringify(query)));
+        });
+
+        it('should accept numbers', async () => {
+          const usableFrom = 1581984000000;
+          const expectedUsableFrom = '20200218T000000Z';
+
+          const signedUrl = await signer.getSignedUrl({
+            version: 'v4',
+            method: 'GET',
+            usableFrom,
+            expires: usableFrom + 5000,
+          });
+
+          const query = {
+            'X-Goog-Date': expectedUsableFrom,
+          };
+          assert(signedUrl.includes(qsStringify(query)));
+        });
+
+        it('should accept strings', async () => {
+          const usableFrom = '12-12-2099';
+          const usableFromDate = new Date(usableFrom);
+          const expectedUsableFrom = '20991212T000000Z';
+
+          const signedUrl = await signer.getSignedUrl({
+            version: 'v4',
+            method: 'GET',
+            usableFrom,
+            expires: usableFromDate.valueOf() + 5000,
+          });
+
+          const query = {
+            'X-Goog-Date': expectedUsableFrom,
+          };
+          assert(signedUrl.includes(qsStringify(query)));
+        });
+
+        it('should throw if a date is invalid', () => {
+          const usableFrom = new Date('31-12-2019');
+
+          assert.throws(() => {
+            signer.getSignedUrl({
+              version: 'v4',
+              method: 'GET',
+              usableFrom,
+              expires: Date.now() + 5000,
+            });
+          }, /The usable from date provided was invalid\./);
+        });
+
+        it('should throw if an expiration date from the before usableFrom date is given', () => {
+          const expires = Date.now() + 5;
```

Comment:
Based on our pairing:

`+ 5` is may not work as expected, could you update the line to increment based on the unit? e.g. second, minute, hour.

Reply:
I have updated it to be 1 day difference. I am not incrementing by unit as to avoid converting to a date and then back to milliseconds. Do you think I should perform this conversion?